packages/gatsby-link/index.d.ts

```diff
@@ -1,11 +1,21 @@
 import * as React from "react";
 
-export const push: (to: LocationDescriptor) => void;
-export const replace: (to: LocationDescriptor) => void;
+export const push: (to: string) => void;
+export const replace: (to: string) => void;
 
 // TODO: Remove navigateTo for Gatsby v3
-export const navigateTo: (to: LocationDescriptor) => void;
+export const navigateTo: (to: string) => void;
```

Comment:
Does typescript have a way to say something is deprecated?

Reply:
Doesn't look like it. TSLint has a rule to warn when it see's a deprecated JSDOC comment but it doesn't apply to third party libraries apparently. 

https://github.com/Microsoft/TypeScript/issues/390

Reply:
@KyleAMathews Using the JSDoc `@deprecated` annotation is fine.